src/pkg/util/env/clean.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package env
+
+import (
+	"strings"
+
+	"github.com/opencontainers/runtime-tools/generate"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+// CleanEnv cleans environment variables before running the container
+func CleanEnv(g *generate.Generator, NoHome bool, IsCleanEnv bool, HomeDest string, environment []string) {
+	for _, env := range environment {
+		e := strings.SplitN(env, "=", 2)
+		if len(e) != 2 {
+			sylog.Verbosef("Can't process environment variable %s", env)
+			continue
+		}
+
+		// Transpose environment
+		if strings.HasPrefix(e[0], "SINGULARITYENV_") {
+			e[0] = strings.TrimPrefix(e[0], "SINGULARITYENV_")
+		} else if IsCleanEnv && (e[0] != "HOME" &&
+			e[0] != "TERM" &&
+			e[0] != "LANG" &&
+			e[0] != "http_proxy" &&
+			e[0] != "https_proxy" &&
+			e[0] != "no_proxy" &&
+			e[0] != "all_proxy" &&
+			e[0] != "ftp_proxy" &&
+			e[0] != "SINGULARITY_CONTAINER" &&
```

Comment:
These two variables still aren't ever being set (`SINGULARITY_CONTAINER` && `SINGULARITY_NAME`). From what I remember, `SINGULARITY_CONTAINER` is image argument passed to the CLI and `SINGULARITY_NAME` is the name of the file. For 
example:

```
singularity run /my/test/image.sif

SINGULARITY_CONTAINER=/my/test/image.sif
SINGULARITY_NAME=image.sif
```

Reply:
and when the image comes from a URI?
e.g
```
singularity run library://alpine:latest
```

Reply:
should we leave `SINGULARITY_NAME` when actually having a name, a.k.a when using instances?

Reply:
@bauerm97 ... look at `/.singularity.d/actions/shell` .... `$SINGULARITY_CONTAINER` is being used in that to set PS1 when bash exists as a shell in the container

Reply:
@ArangoGutierrez 

```
$ /usr/local/singularity/2.6.0/bin/singularity shell docker://centos:7
  [ ... ]
SINGULARITY_NAME=centos:7
SINGULARITY_CONTAINER=centos:7
```

Reply:
they *Should* be different , at least `CONTAINER` should point the path/source, i kind of agree with the name. I would use `NAME` when running instances. but is just my opinion

Reply:
```